atc/api/wallserver/message.go

```diff
@@ -0,0 +1,86 @@
+package wallserver
+
+import (
+	"encoding/json"
+	"github.com/concourse/concourse/atc"
+	"net/http"
+	"time"
+)
+
+func (s *Server) GetWall(w http.ResponseWriter, r *http.Request) {
+	var wall atc.Wall
+
+	logger := s.logger.Session("wall")
+
+	w.Header().Set("Content-Type", "application/json")
+
+	message, err := s.wall.GetMessage()
+	if err != nil {
+		logger.Error("failed-to-get-message", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	wall.Message = message
+
+	expiration, err := s.wall.GetExpiration()
+	if err != nil {
+		logger.Error("failed-to-get-expiration", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if !expiration.IsZero() {
+		wall.ExpiresIn = time.Until(expiration).Round(time.Second).String()
+	}
+
+	err = json.NewEncoder(w).Encode(wall)
+	if err != nil {
+		logger.Error("failed-to-encode-json", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func (s *Server) SetWall(w http.ResponseWriter, r *http.Request) {
+	logger := s.logger.Session("wall")
+
+	var wall atc.Wall
+	err := json.NewDecoder(r.Body).Decode(&wall)
+	if err != nil {
+		logger.Error("failed-to-decode-json", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	err = s.wall.SetMessage(wall.Message)
```

Comment:
Maybe we can have the `SetMessage` be a `SetWall` and combine the inserting of both the message and the expiration in one query? We are trying to have API queries use only one transaction in order to speed up the responses during times where the environment is pegged for connections, but it's definitely not a deal breaker.

Reply:
I'm down for that optimization

Reply:
I think it would be a great idea to simplify the `db.Wall` interface down to `Get` `Set` and `Clear`. Simple and CRUDdy